compiler/src/dmd/statementsem.d

```diff
@@ -279,6 +279,25 @@ Statement statementSemanticVisit(Statement s, Scope* sc)
                 ++i;
                 continue;       // look for errors in rest of statements
             }
+
+            // expand tuple variables in order to attach destruction/exception logic
+            if (auto es = s.isExpStatement())
+            {
+                if (es.exp && es.exp.isDeclarationExp())
+                {
+                    auto de = es.exp.isDeclarationExp();
+                    auto vd = de.declaration.isVarDeclaration();
+                    if (vd && vd.aliasTuple && vd.aliasTuple.objects.length)
+                    {
+                        Statements* a = new Statements();
```

Comment:
This creates an unnecessary allocation. See https://github.com/dlang/dmd/pull/15366/files#diff-11fb0212c55c05ed6e743e6776abb3c56996381760c8725255c622f96f5d9d4bR308 instead.

Reply:
Done, even reduced to just one `insert`. (thanks to #15369)